sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -141,17 +140,34 @@ public AmqpAnnotatedMessage getAmqpAnnotatedMessage() {
      * @return A byte array representing the data.
      */
     public byte[] getBody() {
-        final AmqpBodyType type = amqpAnnotatedMessage.getBody().getBodyType();
-        switch (type) {
+        return getBodyAsBinaryData().toBytes();
+    }
+
+    /**
+     * Gets the actual payload/data wrapped by the {@link ServiceBusMessage}.
+     *
+     * <p>
+     * If the means for deserializing the raw data is not apparent to consumers, a common technique is to make use of
+     * {@link #getApplicationProperties()} when creating the event, to associate serialization hints as an aid to
+     * consumers who wish to deserialize the binary data.
+     * </p>
+     *
+     * @return A {@link BinaryData} representing the data.
+     */
+    public BinaryData getBodyAsBinaryData() {
+        final AmqpBodyType bodyType = amqpAnnotatedMessage.getBody().getBodyType();
+        switch (bodyType) {
             case DATA:
-                return Arrays.copyOf(binaryData, binaryData.length);
+                return ((AmqpDataBody) amqpAnnotatedMessage.getBody()).getDataAsBinaryData().stream().findFirst().get();
```

Comment:
Do we always have to use amqpAnnotatedMessage even for messages that don't need this?

Reply:
We are storing binary payload in AmqpMessage to allow user to get to payload via ServiceBusMessage or via  ServiceBusMessage.getAmqpMessage() along with other Amqp properties.
Not creating would mean at construction time, we will have to duplicate binary payload in two places when we need it. 
We are treating `ServiceBusMessage` is a wrapper over `amqpAnnotatedMessage` and everything is stored in `amqpAnnotatedMessage`. 

This is same as other language is doing.